daemon/ucrednet_test.go

```diff
@@ -0,0 +1,156 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"errors"
+	"net"
+	"path/filepath"
+	sys "syscall"
+
+	"gopkg.in/check.v1"
+)
+
+type ucrednetSuite struct {
+	ucred *sys.Ucred
+	err   error
+}
+
+var _ = check.Suite(&ucrednetSuite{})
+
+func (s *ucrednetSuite) getUcred(fd, level, opt int) (*sys.Ucred, error) {
+	return s.ucred, s.err
+}
+
+func (s *ucrednetSuite) SetUpSuite(c *check.C) {
+	getUcred = s.getUcred
+}
+
+func (s *ucrednetSuite) TearDownTest(c *check.C) {
+	s.ucred = nil
+	s.err = nil
+}
+func (s *ucrednetSuite) TearDownSuite(c *check.C) {
+	getUcred = sys.GetsockoptUcred
+}
+
+func (s *ucrednetSuite) TestAcceptConnRemoteAddrString(c *check.C) {
+	s.ucred = &sys.Ucred{Uid: 42}
+	d := c.MkDir()
+	sock := filepath.Join(d, "sock")
+
+	l, err := net.Listen("unix", sock)
+	c.Assert(err, check.IsNil)
+	defer l.Close()
+
+	go func() {
+		cli, err := net.Dial("unix", sock)
+		c.Assert(err, check.IsNil)
+		cli.Close()
+	}()
+
+	wl := &ucrednetListener{l}
+
+	conn, err := wl.Accept()
+	c.Assert(err, check.IsNil)
+	defer conn.Close()
+
+	c.Check(conn.RemoteAddr().String(), check.Matches, "42:.*")
```

Comment:
@chipaca what I mean is that it would be nice to have an actual roundtrip test ucrenetGetUid(ucrenetAddr.String())